amt/client.py

```diff
@@ -81,19 +81,26 @@ def __init__(self, address, password, username='admin', protocol='http'):
         self.password = password
 
     def post(self, payload, ns=None):
-        resp = requests.post(self.uri,
-                             headers={'content-type':
-                                      'application/soap+xml;charset=UTF-8'},
-                             auth=HTTPDigestAuth(self.username, self.password),
-                             data=payload)
-        resp.raise_for_status()
-        if ns:
-            rv = _return_value(resp.content, ns)
-            if rv == 0:
-                return 0
-            print(pp_xml(resp.content))
-        else:
-            return 0
+        try:
+            resp = requests.post(self.uri,
+	                             headers={'content-type':
+	                                      'application/soap+xml;charset=UTF-8'},
+	                             auth=HTTPDigestAuth(self.username, self.password),
+	                             data=payload,
+	                             timeout=5.0)
```

Comment:
The request timeout is typically just a backstop, and I'd be concerned that 5 seconds might not be enough, How about we make it 30?